src/vm/gcenv.ee.cpp

```diff
@@ -993,3 +993,134 @@ MethodTable* GCToEEInterface::GetFreeObjectMethodTable()
     assert(g_pFreeObjectMethodTable != nullptr);
     return g_pFreeObjectMethodTable;
 }
+
+// These are arbitrary, we shouldn't ever be having confrig keys or values
+// longer than these lengths.
+const size_t MaxConfigKeyLength = 255;
+const size_t MaxConfigValueLength = 255;
+
+bool GCToEEInterface::GetBooleanConfigValue(const char* key, bool* value)
+{
+    CONTRACTL {
+        NOTHROW;
+        GC_NOTRIGGER;
+    } CONTRACTL_END;
+
+    // these configuration values are given to us via startup flags.
+    if (strcmp(key, "gcServer") == 0)
+    {
+        *value = g_heap_type == GC_HEAP_SVR;
+        return true;
+    }
+
+    if (strcmp(key, "gcConcurrent") == 0)
+    {
+        *value = g_IGCconcurrent != 0;
+        return true;
+    }
+
+    if (strcmp(key, "GCRetainVM") == 0)
+    {
+        // GCRetainVM is strange in that it is derived from startup flags
+        // on unixes, but not on windows. This call preserves the current
```

Comment:
It is sort of strange that Windows CoreRun uses ICLRRuntimeHost4 and Unix CoreRun doesn't, though; it means that Unix CoreRun flows through an extra layer of configuration code (translating key/value pairs into start flags) that the Windows one doesn't. I don't know how the CLI hosts the runtime so I'm not sure what it does.